### DISCUSSION TITLE
Protecting private routes with createMiddleware and next js middleware

### DISCUSSION THREAD

**Discussion by mirandajc** [amannn/next-intl#931] [Q&A]
The problem is when I have to join the middleware of next and next intl to intercept some routes that need token to access. When the request.nextUrl.pathname is made to redirect to the registration url it does it perfectly, but in the browser remains the URL I am trying to access, I want to know why this happens and in the case of using NextResponse.redirect(new URL('/signup', request.url)) the styles are broken and the intl middleware does not work.

**Comment by stefnto** [comment]
I'm facing the same problem, except the breaking of styles. Basically the URL stays the same but the page is rewritten to the one it should have been redirected to. Is there a work around or is this they intended behaviour? Because right now im using a mix of return the next intl response and next response redirect, which I think can get improved.
